daisy/step_create_instances.go

```diff
@@ -44,17 +45,22 @@ Loop:
 		case <-tick:
 			resp, err := w.ComputeClient.GetSerialPortOutput(path.Base(i.Project), path.Base(i.Zone), i.Name, port, start)
 			if err != nil {
-				// Instance is stopped or stopping.
+				numErr++
 				status, sErr := w.ComputeClient.InstanceStatus(path.Base(i.Project), path.Base(i.Zone), i.Name)
 				switch status {
 				case "TERMINATED", "STOPPED", "STOPPING":
+					// Instance is stopped or stopping.
 					if sErr == nil {
 						break Loop
 					}
 				}
-				w.LogStepInfo(s.name, "CreateInstances", "Instance %q: error getting serial port: %v", i.Name, err)
-				break Loop
+				if numErr > 10 {
+					w.LogStepInfo(s.name, "CreateInstances", "Instance %q: error getting serial port: %v", i.Name, err)
+					break Loop
+				}
+				continue
 			}
+			numErr++
```

Comment:
Shouldn't this be numErr=0 to reset error counter if no error occured on this tick?

Reply:
Oh wow, yeah, you're completely right. Running the tests again ..

Reply:
What about a unit test?

Reply:
splendid idea! Added.